File: java/cmf-bus-api/src/main/java/cmf/bus/IEnvelopeReceiver.java
package cmf.bus;

/**
 * An interface defining the methods by which a client may register to receive
 * envelopes and unregister for the same.
 * @see IEnvelopeBus
 */
public interface IEnvelopeReceiver extends IDisposable {

	/**
	 * Registers a handler to receive envelopes of a particular kind. 
	 * 
	 * @param registration The {@link IRegistration} object that describes the kind of 
	 * envelopes to be received and provides that handler method that will receive them.
	 * 
	 * @throws Exception
	 */
    void register(IRegistration registration) throws Exception;

    /**
     * Unregisters a handler previously registered.  Any messages already received will continue to be processed.
     * 
     * @param registration The {@link IRegistration} instance previously used to register handler.
     * 
     * @throws Exception
     */
    void unregister(IRegistration registration) throws Exception;
}

File: dotnet/source/cmf.bus/IEnvelopeReceiver.cs
using System;

namespace cmf.bus
{
    /// <summary>
    /// An interface defining the methods by which a client may register to receive 
    /// envelopes and unregister for the same.
    /// </summary>
    public interface IEnvelopeReceiver : IDisposable
    {
        /// <summary>
        ///  Registers a handler to receive envelopes of a particular kind.
        /// </summary>
        /// <param name="registration">The <see cref="IRegistration" /> object 
        /// that describes the kind of envelopes to be received and provides that handler 
        /// method that will receive them. </param>
        /// <exception cref="System.Exception">May throw an exception.</exception>
        void Register(IRegistration registration);

        /// <summary>
        /// Unregisters a handler previously registered.
        /// </summary>
        /// <param name="registration">The <see cref="IRegistration" />
        /// instance previously used to register handler.</param>
        /// <remarks>Any messages already received will continue to be processed.</remarks>
        /// <exception cref="System.Exception">May throw an exception.</exception>
        void Unregister(IRegistration registration);
    }
}
